src/services/Store/store/tokenScanning.slice.ts

```diff
@@ -0,0 +1,133 @@
+import { createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { BigNumber } from 'bignumber.js';
+import { call, put, select, takeLatest } from 'redux-saga/effects';
+
+import { Asset, AssetBalanceObject, Network, StoreAccount, TAddress } from '@types';
+import { generateAssetUUID, generateDeterministicAddressUUID, isSameAddress } from '@utils';
+import { mapAsync } from '@utils/asyncFilter';
+import { eqBy, identity, mergeAll, prop, unionWith } from '@vendor';
+
+import { getAllTokensBalancesOfAccounts, getBaseAssetBalances } from '../BalanceService';
+import { getAccounts, updateAccountAssets } from './account.slice';
+import { getAssets } from './asset.slice';
+import { getNetworks } from './network.slice';
+import { AppState } from './reducer';
+
+export const initialState = {
+  scanning: false
+};
+
+const slice = createSlice({
+  name: 'tokenScanning',
+  initialState,
+  reducers: {
+    scanTokens(
+      state,
+      _action: PayloadAction<{ accounts?: StoreAccount[]; assets?: Asset[] } | undefined>
+    ) {
+      state.scanning = true;
+    },
+    finishTokenScan(state) {
+      state.scanning = false;
+    }
+  }
+});
+
+export const { scanTokens, finishTokenScan } = slice.actions;
+
+export default slice;
+
+/**
+ * Selectors
+ */
+
+export const isScanning = createSelector(
+  (state: AppState) => state.tokenScanning.scanning,
+  identity
+);
+
+/**
+ * Sagas
+ */
+export function* scanTokensSaga() {
+  yield takeLatest(scanTokens.type, scanTokensWorker);
+}
+
+export const getTokens = async (networks: Network[], accounts: StoreAccount[], assets: Asset[]) => {
```

Comment:
Per conversation, separate api calls from transforms in order to test the later.